src/sentry/data_export/processors/discover.py

```diff
@@ -0,0 +1,77 @@
+from __future__ import absolute_import
+
+import logging
+
+from sentry.api.utils import get_date_range_from_params
+from sentry.snuba import discover
+from sentry.models import Group, Project
+
+from ..base import ExportError
+
+logger = logging.getLogger(__name__)
+
+
+class DiscoverProcessor(object):
+    """
+    Processor for exports of discover data based on a provided query
+    """
+
+    def __init__(self, organization_id, discover_query):
+        self.projects = self.get_projects(organization_id, discover_query)
+        self.start, self.end = get_date_range_from_params(discover_query)
+        self.params = {
+            "organization_id": organization_id,
+            "project_id": [project.id for project in self.projects],
+            "start": self.start,
+            "end": self.end,
+        }
+        self.header_fields = discover_query["field"]
+        self.data_fn = self.get_data_fn(
+            fields=discover_query["field"], query=discover_query["query"], params=self.params
+        )
+
+    @staticmethod
+    def get_projects(organization_id, query):
```

Comment:
This function in the endpoints also handles permissions (does this user have access to this project). Does that check still happen somewhere?

Reply:
Unfortunately I haven't moved that logic over yet since it's all in the endpoints and it'd be a lot of copying. Ideally I'd want some `util` that both endpoints/my-task can reach into, but I'm not sure if I should be setting that up.

The only way users would be able to get data on projects they shouldn't have access to would be to curl/Postman the request themselves though, it is checked for browser queries

Reply:
I could start working on that if it's blocking the merge, but I have a separate ticket to address this Jira 👌

Reply:
This will be added in a separate PR -> https://github.com/getsentry/sentry/pull/18229